src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterServiceMonitor.scala

```diff
@@ -0,0 +1,105 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import akka.actor.{Actor, Props, Timers}
+import org.broadinstitute.dsde.workbench.leonardo.Metrics
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.google.GoogleProjectDAO
+import org.broadinstitute.dsde.workbench.leonardo.config.ClusterServiceConfig
+import org.broadinstitute.dsde.workbench.leonardo.dao.{JupyterDAO, WelderDAO}
+import org.broadinstitute.dsde.workbench.leonardo.dao.google.GoogleDataprocDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.DbReference
+import org.broadinstitute.dsde.workbench.leonardo.model.Cluster
+import org.broadinstitute.dsde.workbench.leonardo.monitor.ClusterServiceMonitor.{DetectClusterStatus, JupyterStatus, Status, TimerKey, WelderStatus}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.newrelic.NewRelicMetrics
+
+import scala.concurrent.Future
+
+object ClusterServiceMonitor {
+
+
+  val newRelic: NewRelicMetrics = Metrics.newRelic
+
+  def props(config: ClusterServiceConfig, gdDAO: GoogleDataprocDAO, googleProjectDAO: GoogleProjectDAO, dbRef: DbReference, welderDAO: WelderDAO, jupyterDAO: JupyterDAO, newRelic: NewRelicMetrics): Props = {
+    Props(new ClusterServiceMonitor(config, gdDAO, googleProjectDAO, dbRef, welderDAO, jupyterDAO, newRelic: NewRelicMetrics))
+  }
+
+  sealed trait ClusterServiceMonitorMessage
+  case object DetectClusterStatus extends ClusterServiceMonitorMessage
+  case object TimerKey extends ClusterServiceMonitorMessage
+
+  sealed trait ServiceStatus extends Product with Serializable
+  sealed trait JupyterStatus extends ServiceStatus with Product with Serializable
```

Comment:
I wonder if the status checks could be generalized to other tools besides Jupyter/Welder. For example `ClusterMontitorActor` takes a `Map[ClusterTool, () => Future[Boolean]]` for status checking tools. There may be opportunities for code reuse in `ClusterServiceMonitor`.

See https://github.com/DataBiosphere/leonardo/blob/develop/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisor.scala#L157